src/covermanager/coverproviders.cpp

```diff
@@ -134,7 +134,7 @@ void CoverProviders::RemoveProvider(CoverProvider *provider) {
 
 void CoverProviders::ProviderDestroyed() {
 
-  CoverProvider *provider = static_cast<CoverProvider*>(sender());
+  CoverProvider *provider = dynamic_cast<CoverProvider*>(sender());
```

Comment:
`dynamic_cast` does not work here when `sender()` is destroyed, that's the reason `qobject_cast` wasn't used here in the first place. This will crash.